src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,403 @@
+      geodatabaseButton.setMaxWidth(Double.MAX_VALUE);
+      controlsVBox.getChildren().add(geodatabaseButton);
+
+      // create progress indicator
+      progressIndicator = new ProgressIndicator();
+      progressIndicator.setVisible(false);
+
+      // add listener to handle generate/sync geodatabase button
+      geodatabaseButton.setOnAction(e -> {
+        if (currentEditState == EditState.NOTREADY) {
+          // update button text and disable button
+          geodatabaseButton.setDisable(true);
+          geodatabaseButton.setText("Generating Geodatabase...");
+
+          // show the extent of the geodatabase to be generated
+          final SimpleLineSymbol boundarySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF0000FF, 5);
+          final Envelope extent = mapView.getVisibleArea().getExtent();
+          Graphic boundary = new Graphic(extent, boundarySymbol);
+          graphicsOverlay.getGraphics().add(boundary);
+
+          // generate Geodatabase for the chosen extent
+          generateGeodatabase(extent);
+        } else if (currentEditState == EditState.READY) {
+          syncGeodatabase();
+        }
+      });
+
+      // handle clicks on the map view to select and move features
+      mapView.setOnMouseClicked((event) -> {
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // get screen point where user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+
+          // get map location corresponding to screen point
+          Point mapPoint = mapView.screenToLocation(screenPoint);
+
+          // iterate through all feature layers in the map
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            final FeatureLayer featureLayer = (FeatureLayer) layer;
+
+            // identify any clicked feature
+            ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
+            results.addDoneListener(() -> {
+              try {
+                // get selected feature, if it is a point
+                List<GeoElement> elementList = results.get().getElements();
+                if (elementList.size() > 0 && elementList.get(0) instanceof ArcGISFeature && elementList.get(0).getGeometry().getGeometryType() == GeometryType.POINT) {
+
+                  // clicked on a feature, select it
+                  ArcGISFeature selectedFeature = (ArcGISFeature) elementList.get(0);
+                  // TODO: this doesn't clear the selection in all FL, hence some elements stay selected?
+                  featureLayer.clearSelection(); // clear previous selection
+                  featureLayer.selectFeature(selectedFeature);
+
+                  // change the state to editing
+                  currentEditState = EditState.EDITING;
+
+                } else {
+                  // didn't click on a feature
```

Comment:
I'm a bit confused by this section. I think some clearer commenting would help. If the user didn't click on a feature, why is the selected feature having its geometry updated in this code block? Is this referring more to if the user didn't click on a point? Though the user can't select anything other than a point.

Reply:
It's quite a tricky one, yeah. Maybe the best way to clarify this would be to only use the term feature? 
I've made sure to now only refer to features rather than points (since features can represent points, but not all points are necessarily features). 
In short, we need to find out whether we clicked on a feature or not. If we did, we want to select it (and de-select and previously selected ones, so only one feature is ever selected at any given time). If we didn't click on a feature we want to check whether a feature has already been selected (which means it will now be moved). If that's the case, then we move it. If no feature was selected, we just clicked anywhere else on the map, and we do nothing (which was the other comment you highlighted)